internal/operators/ocs/validations.go

```diff
@@ -113,134 +111,82 @@ func (o *operator) computeResourcesAllNodes(cluster *models.Cluster, nodeInfo *r
 	var status string
 	numAvailableHosts := int64(len(hosts))
 
-	if len(nodeInfo.insufficientHosts) > 0 {
-		for _, hostStatus := range nodeInfo.insufficientHosts {
-			status = status + hostStatus + ".\n"
-		}
-		o.log.Info("Validate Requirements status ", status)
-		return false, status
-	}
-
-	if nodeInfo.numDisks%3 != 0 {
-		status = fmt.Sprint("The number of disks in the cluster for OCS must be a multiple of 3 with a minimum size of ", MinDiskSize, " GB")
+	if ocsClusterResources.numberOfDisks%3 != 0 {
+		status = fmt.Sprint("The number of disks in the cluster for OCS must be a multiple of 3 with a minimum size of ", o.config.OCSMinDiskSize, " GB")
 		o.log.Info(status)
 		return false, status
 	}
 
-	if numAvailableHosts == o.config.OCSRequiredHosts { // for 3 hosts
-		if !validateRequirementsForMode(o, nodeInfo, compactMode) { // check for master nodes requirements
-			status = o.setStatusInsufficientResources(nodeInfo, compactMode)
+	if numAvailableHosts == o.config.OCSNumMinimumHosts { // for 3 hosts
+		if validateDiskAndHostWithDisk(o, ocsClusterResources) {
+			status = o.setStatusInsufficientResources(ocsClusterResources, compactMode)
 			return false, status
 		}
 		o.config.OCSDeploymentType = compactMode
 		status = "OCS Requirements for Compact Mode are satisfied"
 		return true, status
 	}
-
-	if !validateRequirementsForMode(o, nodeInfo, standardMode) { // check for worker nodes requirement
-		if !validateRequirementsForMode(o, nodeInfo, minimalMode) { // check for worker nodes requirements
-			status = o.setStatusInsufficientResources(nodeInfo, minimalMode)
-			return false, status
-		}
-		status = "Requirements for OCS Minimal Deployment are satisfied"
-		o.config.OCSDeploymentType = minimalMode
-		return true, status
+	//TODO: update after mode is decided
+	if validateDiskAndHostWithDisk(o, ocsClusterResources) { // check for worker nodes requirement
+		status = o.setStatusInsufficientResources(ocsClusterResources, minimalMode)
+		return false, status
 	}
-	status = "OCS Requirements for Standard Deployment are satisfied"
-	o.config.OCSDeploymentType = standardMode
+	status = "Requirements for OCS Minimal Deployment are satisfied"
+	o.config.OCSDeploymentType = minimalMode
 	return true, status
 }
 
-func validateRequirementsForMode(o *operator, nodeInfo *resourceInfo, mode ocsDeploymentMode) bool {
-	var totalCPUs int64
-	var totalRAM int64
-
-	if mode == compactMode || mode == standardMode {
-		totalCPUs = o.config.OCSStandardAndCompactModeCPUCount
-		totalRAM = o.config.OCSStandardAndCompactModeRAMGiB
-	} else {
-		totalCPUs = o.config.OCSMinimumCPUCount
-		totalRAM = o.config.OCSMinimumRAMGiB
-	}
-	return nodeInfo.cpuCount >= totalCPUs && nodeInfo.ram >= conversions.GibToBytes(totalRAM) && nodeInfo.numDisks >= o.config.OCSRequiredDisk && nodeInfo.hostsWithDisks >= o.config.OCSRequiredHosts
+func validateDiskAndHostWithDisk(o *operator, ocsClusterResources *ocsClusterResourcesInfo) bool {
```

Comment:
The function name is confusing. I would prefer calling it validateRequirements so we can add additional validations if required in the future. Return true on success and false on failure so reverse the checks.

Reply:
Done